lib/Sema/CSSimplify.cpp

```diff
@@ -6791,22 +6794,25 @@ ConstraintSystem::simplifyCheckedCastConstraint(
     Type toKeyType, toValueType;
     std::tie(toKeyType, toValueType) = *isDictionaryType(toType);
 
-    if (simplifyCheckedCastConstraint(fromKeyType, toKeyType,
-                                      subflags | TMF_ApplyingFix,
-                                      locator) == SolutionKind::Error)
+    auto elementLocator =
+        locator.withPathElement(LocatorPathElt::KnownStdlibCollectionElement());
+    if (simplifyCheckedCastConstraint(fromKeyType, toKeyType, subflags,
```

Comment:
It took me a bit to understand what the problem is in this case - `formUnsolved` drops `flags` so `TMF_ApplyingFix` doesn't get propagated to the new constraint...

Reply:
Ah sorry, I think I could have done better on the description, but yeah is exactly that :)

Reply:
No worries, it's usually helpful to explain what happens incorrectly instead of stating that something didn't work :)

Reply:
Thanks!